zookeeper-server/src/test/java/org/apache/zookeeper/server/quorum/QuorumPeerMainTest.java

```diff
@@ -436,6 +438,35 @@ public void testElectionFraud() throws IOException, InterruptedException {
         Assert.assertTrue("falseLeader never rejoins the quorum", foundFollowing);
     }
 
+    @Test
+    public void testBadMyId() throws Exception {
+        ClientBase.setupTestEnv();
+
+        final int CLIENT_PORT_QP1 = PortAssignment.unique();
+        List<Integer> invalidMyidList = new ArrayList<>();
+        invalidMyidList.add(-1);
```

Comment:
You should consider using JUnit's parameterised test feature to run the same test with multiple myids.